py/desispec/preproc.py

```diff
@@ -317,7 +320,11 @@ def get_calibration_image(cfinder,keyword,entry) :
     elif keyword == "MASK" :
         return read_mask(filename=filename)
     elif keyword == "DARK" :
-        return read_dark(filename=filename)
+        if header is not None: # Use master bias+dark file if header is set to pass exptime
```

Comment:
Same as other comment. Having a non null header argument to trigger a different read routine seem fragile. I would consider another keyword for this (like 'DARKARRAY').

Reply:
I have added a new keyword "DARKNL" which means non-linear